pyhanko/pdf_utils/reader.py

```diff
@@ -204,10 +204,8 @@ def __init__(self, stream, strict: bool = True):
 
     @property
     def security_handler(self):
-        if self._security_handler is None:
-            encrypt_dict = self._get_encryption_params()
-            if encrypt_dict is not None:
-                self._security_handler = SecurityHandler.build(encrypt_dict)
+        if self.encrypt_dict and not self._security_handler:
+            self._security_handler = SecurityHandler.build(self.encrypt_dict)
```

Comment:
Otherwise if calling self.encrypt_dict is an issue:

```python
@property
  def security_handler(self):
      if (encrypt_dict := self.encrypt_dict) and not self._security_handler:
          self._security_handler = SecurityHandler.build(encrypt_dict)
      return self._security_handler
```